plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraTable.java

```diff
@@ -0,0 +1,193 @@
+            List<ColumnDefinition> columnDefinitions,
+            Optional<String> tableOptions)
+    {
+        this(sqlExecutor, server, keyspace, namePrefix, columnDefinitions, tableOptions, ImmutableList.of());
+    }
+
+    public TestCassandraTable(
+            SqlExecutor sqlExecutor,
+            CassandraServer server,
+            String keyspace,
+            String namePrefix,
+            List<ColumnDefinition> columnDefinitions,
+            List<String> rowsToInsert)
+    {
+        this(sqlExecutor, server, keyspace, namePrefix, columnDefinitions, Optional.empty(), rowsToInsert);
+    }
+
+    public TestCassandraTable(
+            SqlExecutor sqlExecutor,
+            CassandraServer server,
+            String keyspace,
+            String namePrefix,
+            List<ColumnDefinition> columnDefinitions,
+            Optional<String> tableOptions,
+            List<String> rowsToInsert)
+    {
+        super(sqlExecutor, keyspace + "." + namePrefix, tableDefinition(columnDefinitions, tableOptions), columnNames(columnDefinitions), rowsToInsert);
+        this.tableName = getName().substring(keyspace.length() + 1);
+
+        try {
+            server.refreshSizeEstimates(keyspace, tableName);
+        }
+        catch (Exception e) {
+            try (TestCassandraTable ignored = this) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    public String getTableName()
+    {
+        return tableName;
+    }
+
+    public String getTableNameInKeyspace()
+    {
+        return getName();
+    }
+
+    @Override
+    protected String createTableName(String namePrefix)
+    {
+        // Override because Cassandra table names are limited to 48 characters in size
+        // https://cassandra.apache.org/doc/trunk/cassandra/cql/ddl.html#common-definitions
+        String tableNameInKeyspace = namePrefix + randomTableSuffix();
+        return tableNameInKeyspace.substring(0, min(tableNameInKeyspace.length(), tableNameInKeyspace.indexOf(".") + 1 + TABLE_NAME_MAXIMUM_LENGTH));
+    }
+
+    private static String tableDefinition(List<ColumnDefinition> columnDefinitions, Optional<String> tableOptions)
+    {
+        ImmutableList.Builder<String> partitionColumns = ImmutableList.builder();
+        ImmutableList.Builder<String> clusterColumns = ImmutableList.builder();
+        for (ColumnDefinition columnDefinition : columnDefinitions) {
+            if (columnDefinition.primaryKeyType == PrimaryKeyType.PARTITION) {
+                partitionColumns.add(columnDefinition.name);
+            }
+            else if (columnDefinition.primaryKeyType == PrimaryKeyType.CLUSTER) {
+                clusterColumns.add(columnDefinition.name);
+            }
+        }
+        List<String> partitions = partitionColumns.build();
+        List<String> clusters = clusterColumns.build();
+
+        checkState(partitions.size() > 0, "Cassandra table must have at least one partition key");
```

Comment:
nit: `checkState` is generally to check "state" of an object. For local verifications `verify` is the more appropriate method.